src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -0,0 +1,1054 @@
+#include "expression_evaluator.hpp"
+
+#include <iterator>
+#include <type_traits>
+
+#include "boost/variant/apply_visitor.hpp"
+
+#include "all_parameter_variant.hpp"
+#include "expression/abstract_expression.hpp"
+#include "expression/abstract_predicate_expression.hpp"
+#include "expression/arithmetic_expression.hpp"
+#include "expression/binary_predicate_expression.hpp"
+#include "expression/case_expression.hpp"
+#include "expression/exists_expression.hpp"
+#include "expression/expression_factory.hpp"
+#include "expression/extract_expression.hpp"
+#include "expression/function_expression.hpp"
+#include "expression/in_expression.hpp"
+#include "expression/list_expression.hpp"
+#include "expression/logical_expression.hpp"
+#include "expression/pqp_column_expression.hpp"
+#include "expression/pqp_select_expression.hpp"
+#include "expression/value_expression.hpp"
+#include "expression_functors.hpp"
+#include "like_matcher.hpp"
+#include "operators/abstract_operator.hpp"
+#include "resolve_type.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "sql/sql_query_plan.hpp"
+#include "storage/materialize.hpp"
+#include "storage/value_column.hpp"
+#include "utils/assert.hpp"
+#include "utils/lambda_visitor.hpp"
+
+using namespace std::string_literals;
+using namespace opossum::expression_factory;
+
+namespace opossum {
+
+ExpressionEvaluator::ExpressionEvaluator(const std::shared_ptr<const Table>& table, const ChunkID chunk_id)
+    : _table(table), _chunk(_table->get_chunk(chunk_id)) {
+  _output_row_count = _chunk->size();
+  _column_materializations.resize(_chunk->column_count());
+}
+
+template <typename R>
+std::shared_ptr<ExpressionResult<R>> ExpressionEvaluator::evaluate_expression_to_result(
+    const AbstractExpression& expression) {
+  switch (expression.type) {
+    case ExpressionType::Arithmetic:
+      return _evaluate_arithmetic_expression<R>(static_cast<const ArithmeticExpression&>(expression));
+
+    case ExpressionType::Logical:
+      return _evaluate_logical_expression<R>(static_cast<const LogicalExpression&>(expression));
+
+    case ExpressionType::Predicate:
+      return _evaluate_predicate_expression<R>(static_cast<const AbstractPredicateExpression&>(expression));
+
+    case ExpressionType::Select: {
+      const auto* pqp_select_expression = dynamic_cast<const PQPSelectExpression*>(&expression);
+      Assert(pqp_select_expression, "Can only evaluate PQPSelectExpression");
```

Comment:
You could add a comment that you don't want LQP expressions here.

Reply:
added `LQPSelectExpressions need to be translated first`